framework/CustomModes/MPILegacySimulationMode.py

```diff
@@ -201,7 +201,9 @@ def modifyInfo(self, runInfoDict):
     newRunInfo['precommand'] = runInfoDict["MPIExec"]+" "+nodeCommand+" -n "+str(numMPI)+" "+runInfoDict['precommand']
     if(runInfoDict['NumThreads'] > 1):
       #add number of threads to the post command.
-      newRunInfo['postcommand'] = " --n-threads=%NUM_CPUS% "+runInfoDict['postcommand']
+      thCommand = runInfoDict["threadCommand"].strip()
+      thCommand = threadCommand.replace("%#%","") if "%#%" in threadCommand else "{} ".format(threadCommand)
+      newRunInfo['postcommand'] =" {}%NUM_CPUS% {}".format(thCommand,runInfoDict['postcommand'])
```

Comment:
What about defining the default thread parameter: " --n-threads=%NUM_CPUS% " and changing this to:
```python
newRunInfo['postcommand'] =" {} {}".format(threadCommand,runInfoDict['postcommand'])
```

Reply:
wouldn't this approach overwrite the "%NUM_CPUS%" directive?

Reply:
So my proposal is go from:
```python
  newRunInfo['postcommand'] = " --n-threads=%NUM_CPUS% "+runInfoDict['postcommand']
```
to (with threadCommand =  " --n-threads=%NUM_CPUS% ")
```python
 newRunInfo['postcommand'] =" {} {}".format(threadCommand,runInfoDict['postcommand'])
```
So by default this would be equivalent to:
```python
 newRunInfo['postcommand'] ="  --n-threads=%NUM_CPUS% {}".format(runInfoDict['postcommand'])
 ```
 If on the other hand threadCommand was set to say " -j %NUM_CPUS% " then this would be equivalent to:
 ```python
  newRunInfo['postcommand'] ="  -j %NUM_CPUS% {}".format(runInfoDict['postcommand'])
 ```
 So I don't think this would overwrite the %NUM_CPUS% directive.